src/aead/aes_gcm.rs

```diff
@@ -129,4 +127,56 @@ mod tests {
         test_aead(&aead::AES_256_GCM,
                   "crypto/cipher/test/aes_256_gcm_tests.txt");
     }
+
+    fn convert_block(data: &[u8]) -> [u8; AES_BLOCK_SIZE] {
+        let mut buf = [0u8; AES_BLOCK_SIZE];
+        buf.clone_from_slice(data);
+        buf
+    }
+
+    #[test]
+    pub fn test_aes() {
+        test::from_file("src/aead/aes_tests.txt", |section, test_case| {
+            assert_eq!(section, "");
+            let key = test_case.consume_bytes("Key");
+            let input = convert_block(&test_case.consume_bytes("Input"));
+            let output = convert_block(&test_case.consume_bytes("Output"));
+
+            let mut aes_key = AesKey {
+                rd_key: [0u32; 4 * (AES_MAX_ROUNDS + 1)],
+                rounds: 0,
+            };
+
+            unsafe {
+                // Key setup
+                if AES_set_encrypt_key(key.as_ptr(), key.len() * 8, &mut aes_key) != 0 {
+                    panic!("Failed to set aes key");
+                }
+
+                // Encryption into separate buffer
+                let mut buf = [0u8; AES_BLOCK_SIZE];
+                AES_encrypt(input.as_ptr(), buf.as_mut_ptr(), &aes_key);
+                assert_eq!(buf, output);
+
+                // In-place encryption
+                buf.clone_from_slice(&input[..]);
+                AES_encrypt(buf.as_ptr(), buf.as_mut_ptr(), &aes_key);
+                assert_eq!(buf, output);
+            };
+            Ok(())
+        })
+    }
+
+    const AES_BLOCK_SIZE: usize = 16;
+
+    #[repr(C)]
+    pub struct AesKey {
```

Comment:
The conventions in *ring* are:

* Use the same name as the C type. In this case, `AES_KEY`. You'd need to do `#[allow(non_camel_case_types)]`.
* Add a comment "// Keep this in sync with `AES_KEY` in aes.h"
* Add a similar comment above AES_KEY in aes.h saying "/* Keep this in sync with AES_KEY in aes_gcm.rs. */